### DISCUSSION TITLE
Need to improve missing parts in documentation - Installation

### DISCUSSION THREAD

**Discussion by eroscai** [SwiftGen/SwiftGen#1011] [Q&A]
Recently, when i collaborated with another college to develop, it often occurred that the generated file were automatically modified(Usually the order of the fields is changing). 
So we tried to find a solution, and finally found that the generated files need to ignored in the project, which requires following steps:
1. Drag the new `Run Script` phase **above** the `Compile Sources` phase and **below** `Check Pods Manifest.lock`
2. Add `$SRCROOT/your_generated.swift` to the "Output Files" of the Build Phase
3. Uncheck "Based on dependency analysis" so that SwiftGen is run on each build
4. If the generated file has been added to the project before, it needs to be deleted in the file directory. Please keep the reference relationship in the project
5. If you haven't added before, build your project, in Finder you will now see `your_generated.swift` in the `$SRCROOT` folder, drag these `your_generated.swift` files into your project and **uncheck** `Copy items if needed`
6. Add these *your_generated.swift pattern to your `.gitignore` file to prevent unnecessary conflicts.

The above steps ensure that i can use SwiftGen properly in regular builds or CI builds, I think it need to be shown in the relevant documentation.

**Comment by djbe** [comment]
1. Correct. Seems pretty logical TBH (it generates source code, obviously that needs to be compiled)
2. There's a `config generate-xcfilelists` command to generate the input & output file lists, so you don't need to do this manually.
3. Don't do that. See item 2 above, use those file lists.
4. No need to delete any files?
5. I'd recommend running SwiftGen from terminal once to generate all the output files, so you can easily add them to your project.
6. I'd recommend against this, we commit our generated files for example. It allows us to keep track of changes to generated code, and avoid having to run it on CI for example.

**Reply by eroscai** [reply]
> 1. Correct. Seems pretty logical TBH (it generates source code, obviously that needs to be compiled)
> 2. There's a `config generate-xcfilelists` command to generate the input & output file lists, so you don't need to do this manually.
> 3. Don't do that. See item 2 above, use those file lists.
> 4. No need to delete any files?
> 5. I'd recommend running SwiftGen from terminal once to generate all the output files, so you can easily add them to your project.
> 6. I'd recommend against this, we commit our generated files for example. It allows us to keep track of changes to generated code, and avoid having to run it on CI for example.

The key point here is why the generated file need to be ignored. As mentioned above, the files generated by different people's environments will have different results during our use. Specially, there are no resource updated, but the newly generated file has changed with fields order, like below:
<img width="796" alt="image" src="https://user-images.githubusercontent.com/280369/187332659-a31dcdd6-a4e4-4fe3-91ed-6bbf4000134c.png">

So why does this problem happen and how to avoid it?

**Reply by djbe** [reply]
That seems like a bug (would need more info about your setup). Please check the existing issues, such as https://github.com/SwiftGen/SwiftGen/issues/891, and give the related PRs such as https://github.com/SwiftGen/SwiftGen/pull/890 a try.

Again, would need more info about your setup, and an actual test case.
